static/app/views/releases/detail/overview/sidebar/thresholdStatuses.tsx

```diff
@@ -0,0 +1,114 @@
+import {useCallback, useEffect, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Client} from 'sentry/api';
+import * as SidebarSection from 'sentry/components/sidebarSection';
+import {IconCheckmark, IconWarning} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import {Organization, Release, ReleaseProject} from 'sentry/types';
+import {getExactDuration} from 'sentry/utils/formatters';
+import useApi from 'sentry/utils/useApi';
+
+import {
+  ReleaseDetailsTable,
+  ReleaseDetailsTableRow,
+} from '../../../components/releaseDetailsSideTable';
+import {fetchThresholdStatuses} from '../../../utils/fetchThresholdStatus';
+import {ThresholdStatus, ThresholdStatusesQuery} from '../../../utils/types';
+
+type Props = {
+  organization: Organization;
+  project: Required<ReleaseProject>;
+  release: Release;
+  selectedEnvs: string[];
+};
+
+function ThresholdStatuses({project, release, organization, selectedEnvs}: Props) {
+  const api: Client = useApi();
+  const [thresholdStatuses, setThresholdStatuses] = useState<ThresholdStatus[]>([]);
+  const fetchThresholdStatusCallback = useCallback(async () => {
+    const fuzzSec = 30;
+    const start = new Date(new Date(release.dateCreated).getTime() - fuzzSec * 1000);
+    const end = new Date(new Date(release.dateCreated).getTime() + fuzzSec * 1000);
+
+    const releaseVersion: string = release.version;
+
+    const query: ThresholdStatusesQuery = {
+      start: start.toISOString(),
+      end: end.toISOString(),
+      release: [releaseVersion],
```

Comment:
can you remind me why we decided to go with querying within a start/end time and calculating whether threshold statuses are healthy ad hoc rather than storing threshold them in the database?

Reply:
This make it easier to calculate on the fly for polling

eventually we'll want to figure out storing the health on some data model if the window is past as well